test/test_rosbag/bag_migration_tests/CMakeLists.txt

```diff
@@ -5,47 +5,136 @@
+    PartiallyMigrated.msg
+    Renamed1.msg
+    Simple.msg
+    SimpleMigrated.msg
+    SubUnmigrated.msg
+    Unmigrated.msg
+  )
+elseif(${MSG_DIRECTORY} STREQUAL "msg_gen2")
+  add_message_files(
+    DIRECTORY ${MSG_DIRECTORY}
+    FILES
+    Constants.msg
+    Converged.msg
+    Convergent.msg
+    MigratedAddSub.msg
+    MigratedExplicit.msg
+    MigratedImplicit.msg
+    MigratedMixed.msg
+    PartiallyMigrated.msg
+    Renamed2.msg
+    Simple.msg
+    SimpleMigrated.msg
+    SubUnmigrated.msg
+    Unmigrated.msg
+  )
+elseif(${MSG_DIRECTORY} STREQUAL "msg_gen3")
+  add_message_files(
+    DIRECTORY ${MSG_DIRECTORY}
+    FILES
+    Converged.msg
+    MigratedAddSub.msg
+    MigratedExplicit.msg
+    MigratedImplicit.msg
+    MigratedMixed.msg
+    PartiallyMigrated.msg
+    Renamed3.msg
+    Simple.msg
+    SimpleMigrated.msg
+    SubUnmigrated.msg
+    Unmigrated.msg
+  )
+elseif(${MSG_DIRECTORY} STREQUAL "msg_gen4")
+  add_message_files(
+    DIRECTORY ${MSG_DIRECTORY}
+    FILES
+    Constants.msg
+    Converged.msg
+    MigratedAddSub.msg
+    MigratedExplicit.msg
+    MigratedImplicit.msg
+    MigratedMixed.msg
+    PartiallyMigrated.msg
+    Renamed4.msg
+    Simple.msg
+    SimpleMigrated.msg
+    SubUnmigrated.msg
+    Unmigrated.msg
+  )
+elseif(${MSG_DIRECTORY} STREQUAL "msg_current")
+  add_message_files(
+    DIRECTORY ${MSG_DIRECTORY}
+    FILES
+    Constants.msg
+    Converged.msg
+    MigratedAddSub.msg
+    MigratedExplicit.msg
+    MigratedImplicit.msg
+    MigratedMixed.msg
+    PartiallyMigrated.msg
+    Renamed5.msg
+    Simple.msg
+    SimpleMigrated.msg
+    SubUnmigrated.msg
+    Unmigrated.msg
+  )
+else()
+  message(FATAL_ERROR "Messages directory '${MSG_DIRECTORY}' does NOT exist!")
+endif()
+
 generate_messages(DEPENDENCIES rosgraph_msgs std_msgs)
 
 catkin_download_test_data(download_data_test_constants_gen1.bag http://download.ros.org/data/test_rosbag/constants_gen1.bag FILENAME test/constants_gen1.bag MD5 77ec8cb20e823ee3f3a87d07ea1132df )
 catkin_download_test_data(download_data_test_constants_gen2.bag http://download.ros.org/data/test_rosbag/constants_gen2.bag FILENAME test/constants_gen2.bag MD5 4efd8459bf9403d1171dff06abcd6548 )
 catkin_download_test_data(download_data_test_converged_gen1.bag http://download.ros.org/data/test_rosbag/converged_gen1.bag FILENAME test/converged_gen1.bag MD5 8e3524157d31b5761ac951fe16e03e12 )
 catkin_download_test_data(download_data_test_converged_gen2.bag http://download.ros.org/data/test_rosbag/converged_gen2.bag FILENAME test/converged_gen2.bag MD5 0ad4041d2e3bab8262c12020ec3e048e )
 catkin_download_test_data(download_data_test_converged_gen3.bag http://download.ros.org/data/test_rosbag/converged_gen3.bag FILENAME test/converged_gen3.bag MD5 90dd16cea5c51fca65be617654fb6b76 )
+#catkin_download_test_data(download_data_test_converged_gen4.bag http://download.ros.org/data/test_rosbag/converged_gen4.bag FILENAME test/converged_gen4.bag MD5 c88ac1f4a9a5eebd667ec55224988e40 )
```

Comment:
Thanks. PR updated to use them.

Also rebased/updated #1011 that should pass the test; remember that this PR (#1010 it's only here to show that the current code fails to handle the new test provided)